disnake/ext/commands/params.py

```diff
@@ -1058,13 +1151,58 @@ def Param(
 param = Param
 
 
-def inject(function: Callable[..., Any]) -> Any:
+def inject(function: Callable[..., Any], *, autocompleters: Dict[str, Callable] = None) -> Any:
     """A special function to use the provided function for injections.
     This should be assigned to a parameter of a function representing your slash command.
 
     .. versionadded:: 2.3
+
+    .. versionchanged:: 2.6
+        Added ``autocompleters`` keyword-only argument.
+
+    Parameters
+    ----------
+    function: Callable
+        The injection function.
+    autocompleters: Dict[:class:`str`, Callable]
+        A mapping of the injection's option names to their respective autocompleters.
+
+        See also :func:`Injection.autocomplete`.
+
+        .. versionadded:: 2.6
+
+    Returns
+    -------
+    :class:`Injection`
+        The resulting injection
     """
-    return Injection(function)
+
+    return Injection(function, autocompleters=autocompleters)
+
+
+def injection(*, autocompleters: Dict[str, Callable] = None) -> Callable[[Callable[..., Any]], Any]:
```

Comment:
unspecific typehints, see the other comment by shift: https://github.com/DisnakeDev/disnake/pull/675/files#r961160293